unifiedpush_android/android/src/main/kotlin/org/unifiedpush/flutter/connector/Plugin.kt

```diff
@@ -20,85 +21,75 @@ class Plugin : ActivityAware, FlutterPlugin, MethodCallHandler {
-         * 3. registerApp() register the end user application to the distributor
-         */
         @JvmStatic
         private fun getDistributors(context: Context,
                                     result: Result?){
-            val distributors = up.getDistributors(context)
-            result?.success(distributors)
-        }
-
-        @JvmStatic
-        private fun getDistributor(context: Context,
-                                    result: Result?){
-            val distributor = up.getDistributor(context)
-            result?.success(distributor)
-        }
+            val intent = Intent()
+            intent.action = ACTION_REGISTER
+            val distributors = context.packageManager.queryBroadcastReceivers(intent, 0).mapNotNull {
+                if (it.activityInfo.exported || it.activityInfo.packageName == context.packageName) {
+                    val packageName = it.activityInfo.packageName
+                    Log.d("UP-Registration", "Found distributor with package name $packageName")
+                    packageName
+                } else {
+                    null
+                }
+            }
 
-        @JvmStatic
-        private fun saveDistributor(context: Context,
-                                    args: ArrayList<*>?,
-                                    result: Result?){
-            val distributor = args!![0] as String
-            up.saveDistributor(context,distributor)
-            result?.success(true)
+            result?.success(distributors)
         }
 
         @JvmStatic
         private fun registerApp(context: Context,
                                 args: ArrayList<*>?,
                                 result: Result?){
-            val instance: String = (args?.get(0) ?: "") as String
-            if (instance.isEmpty()) {
-                up.registerApp(context)
-            } else {
-                up.registerApp(context, instance)
-            }
+            val distributor = args!![0] as String
+            val token = args!![1] as String
+
+            val broadcastIntent = Intent()
+            broadcastIntent.`package` = distributor
+            broadcastIntent.action = ACTION_REGISTER
+            broadcastIntent.putExtra(EXTRA_TOKEN, token)
+            broadcastIntent.putExtra(EXTRA_APPLICATION, context.packageName)
+            context.sendBroadcast(broadcastIntent)
+
+            context.getSharedPreferences(PREF_TOKEN_DISTRIB_MAP, Context.MODE_PRIVATE)
+            .edit()
+            .putString(token, distributor)
+            .apply()
+
             result?.success(true)
         }
 
         @JvmStatic
         private fun unregister(context: Context,
                                args: ArrayList<*>?,
                                result: Result) {
-            val instance: String = (args?.get(0) ?: "") as String
-            if (instance.isEmpty()) {
-                up.unregisterApp(context)
+            val token = args!![1] as String
```

Comment:
Shouldn't this be 0?

Reply:
This is resolved in https://github.com/UnifiedPush/flutter-connector/pull/56#issuecomment-1002779290